File: src/backend/GetToKnowGame/Repositories/PlayerRepository.cs
using GetToKnowGame.Models;
using MongoDB.Driver;

namespace GetToKnowGame.Repositories
{
    public class PlayerRepository : BaseRepository<Player>, IPlayerRepository
    {
        public PlayerRepository(IMongoDatabase database) : base(database, "players")
        {
        }
    }
}

File: src/backend/get-to-know-game-go/repositories/player_repository.go
package repositories

import (
	"get-to-know-game-go/models"

	"go.mongodb.org/mongo-driver/mongo"
)

// PlayerRepositoryImpl implements PlayerRepository
type PlayerRepositoryImpl struct {
	*BaseRepository[models.Player]
}

// NewPlayerRepository creates a new player repository
func NewPlayerRepository(collection *mongo.Collection) PlayerRepository {
	return &PlayerRepositoryImpl{
		BaseRepository: NewBaseRepository[models.Player](collection),
	}
}
